### ISSUE TITLE
api 'public_get_directory_listing' returns redundant struct if no such path

### ISSUE THREAD

**Issue by DrXyzzy** [sagemathinc/cocalc#2370]
Example of value returned from api call when path requested is "nonexist":
```
{'event': 'public_directory_listing',
 'id': '217b4e1b-2599-4185-ae44-e6b5271ba7ef',
 'result': {'error': "command 'cc-ls' (args=/home/user/nonexist) exited with "
                     "nonzero code 1 -- stderr='error: no such path "
                     "'/home/user/nonexist''",
            'files': {'files': []}}}
```

Last line should be:
```
            'files': []}}
```

**Comment by DrXyzzy** [comment]
Output structure looks ok if public path requested exists and is empty:
```
{'event': 'public_directory_listing',
 'id': 'b6df1763-d79b-44dc-88b2-2c6982a33c9e',
 'result': {'files': []}}
```

And looks ok if public path requested exists and is not empty:
```
{'event': 'public_directory_listing',
 'id': 'eb9a2763-8604-460a-b219-f084bba2321c',
 'result': {'files': [{'mtime': 1505191879, 'name': 'hello.txt', 'size': 6}]}}
```

**Comment by DrXyzzy** [comment]
Fixed by #2371.
